src/main/java/pl/byczazagroda/trackexpensesappbackend/service/UserServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package pl.byczazagroda.trackexpensesappbackend.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import pl.byczazagroda.trackexpensesappbackend.dto.AuthRegisterDTO;
+import pl.byczazagroda.trackexpensesappbackend.model.User;
+import pl.byczazagroda.trackexpensesappbackend.model.UserStatus;
+import pl.byczazagroda.trackexpensesappbackend.repository.UserRepository;
+
+import javax.validation.Valid;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@RequiredArgsConstructor
+@Service
+@Validated
+public class UserServiceImpl implements UserService{
+    private final UserRepository userRepository;
+    @Override
+    public void registerUser(@Valid AuthRegisterDTO authRegisterDTO) {
+        if (userRepository.existsByEmail(authRegisterDTO.email())) {
+            throw new IllegalArgumentException("A user with the same email already exists");
```

Comment:
You should change the exception to the `AppRuntimeException` according to our error handling strategy u can refer to other services. It applies to whole service.